syntax = "proto3";

import "tensorflow/compiler/xla/xla_data.proto";

package xla;

// Backend specific HloInstruction config
message PoplarBackendConfig {

  // Instruction config for a fusion instruction
  message FusionConfig {
    // Convolutions
    Window window = 1;
    ConvolutionDimensionNumbers dimension_numbers = 2;
    int64 feature_group_count = 3;
    int64 batch_group_count = 4;

  };

  // Instruction config for a Call repeat instruction
  message RepeatConfig {
    bool is_repeat_loop = 1;
    int64 repeat_count = 2;
  }

  oneof Config {
    FusionConfig fusion_config = 1;
    RepeatConfig repeat_config = 2;
  }

};
